usaspending_api/data_load/fpds_loader.py

```diff
@@ -0,0 +1,340 @@
+
+            inserted_recipients = _insert_recipients(cursor, load_objects)
+            for index in range(0, len(inserted_recipients)):
+                load_objects[index]["transaction_normalized"]["recipient_id"] = inserted_recipients[index][0]
+                load_objects[index]["award"]["recipient_id"] = inserted_recipients[index][0]
+
+            inserted_place_of_performance = _insert_place_of_performance(cursor, load_objects)
+            for index in range(0, len(inserted_place_of_performance)):
+                load_objects[index]["transaction_normalized"][
+                    "place_of_performance_id"
+                ] = inserted_place_of_performance[index][0]
+                load_objects[index]["award"]["place_of_performance_id"] = inserted_place_of_performance[index][0]
+
+            # Handle transaction-to-award relationship for each transaction to be loaded
+            for load_object in load_objects:
+
+                # AWARD GET OR CREATE
+                award_id = _lookup_award_by_transaction(cursor, load_object)
+                if not award_id:
+                    # If there is no award, we need to create one
+                    award_id = _insert_award_by_transaction(cursor, load_object)
+
+                load_object["transaction_normalized"]["award_id"] = award_id
+                ids_of_awards_created_or_updated.add(award_id)
+
+                # TRANSACTION UPSERT
+                transaction_id = _lookup_existing_transaction(cursor, load_object)
+                if transaction_id:
+                    # Inject the Primary Key of transaction_normalized+transaction_fpds that was found, so that the
+                    # following updates can find it to update
+                    load_object["transaction_fpds"]["transaction_id"] = transaction_id
+                    _update_fpds_transaction(cursor, load_object, transaction_id)
+                else:
+                    # If there is no transaction we create a new one.
+                    transaction_id = _insert_fpds_transaction(cursor, load_object)
+
+                load_object["transaction_fpds"]["transaction_id"] = transaction_id
+                load_object["award"]["latest_transaction_id"] = transaction_id
+
+                # No matter what, we need to go back and update the award's latest transaction to the award we just made
+                _update_award_latest_transaction(cursor, award_id, transaction_id)
+
+    return list(ids_of_awards_created_or_updated)
+
+
+def _insert_recipient_locations(cursor, load_objects):
+    sql_to_execute = ""
+    columns, values = setup_mass_load_lists(load_objects, "recipient_location")
+    recipient_location_sql = "INSERT INTO references_location {} VALUES {} RETURNING location_id;".format(
+        columns, values
+    )
+    sql_to_execute += recipient_location_sql
+
+    cursor.execute(sql_to_execute)
+    return cursor.fetchall()
```

Comment:
It would be cleaner to return a list of `location_id` values instead of the cursor's list of tuples

Reply:
but this allows for cleaner patching for the testing. When pulling these out to another file I might write a wrapper around this one in order to get the best of both worlds.